cassiopeia/data/CassiopeiaTree.py

```diff
@@ -1053,28 +1193,141 @@ def relabel_nodes(self, relabel_map: Dict[str, str]) -> None:
     def __register_data_with_tree(self) -> None:
         """Makes the leaf data consistent with the leaves in the tree.
 
-        Removes any leaves from the character matrix, cell metadata, and 
+        Removes any leaves from the character matrix, cell metadata, and
         dissimilarity maps that do not appear in the tree.
+        Additionally, adds any leaves that appear in the tree but not in the
+        character matrix, cell metadata, or dissimilarity map with default values.
+        The default values for each table is as follows:
+        * character matrix: all states are missing values (``missing_state_indicator``)
+        * cell metadata: None
+        * dissimilarity map: ``np.inf`` distance from the leaf to all other leaves
         """
+        leaves_set = set(self.leaves)
         if self.__current_character_matrix is not None:
-            remove_from_charater_matrix = set(self.__current_character_matrix.index) - set(self.leaves)
-            self.__current_character_matrix = self.__current_character_matrix.drop(index = remove_from_charater_matrix)
-                
+            remove_from_charater_matrix = (
+                set(self.__current_character_matrix.index) - leaves_set
+            )
+            self.__current_character_matrix = (
+                self.__current_character_matrix.drop(
+                    index=remove_from_charater_matrix
+                )
+            )
+
+            add_to_character_matrix = leaves_set - set(
+                self.__current_character_matrix.index
+            )
+            for to_add in add_to_character_matrix:
+                # Note that we initialize state each iteration so that each state
+                # is a new object.
+                state = [self.missing_state_indicator] * self.n_character
+                self.__set_character_states(to_add, state)
+                self.__current_character_matrix.loc[to_add] = state
```

Comment:
I think perhaps it'd be more clear to refer to it as a "character vector" rather than "state", as "state" already carries the meaning of an indel